### ISSUE TITLE
Passing outlet references down the template chain

### ISSUE THREAD

**Issue by micpringle** [schibsted/layout#78]
It appears as though you can't pass outlet references to nested templates in the same way you can parameters, but this would definitely be useful in scenarios where a section of a top-level template has been refactored into a separate template so it can be reused multiple times, but that nested template needs access to outlets defined on the top-level view controller.

Here's an example use case where this would be useful...

**Claims.xml**
```xml
<UIView>
  <UIView
    template="Contact.xml"
    top="previous.bottom + 8.0"
    text="In the UK"
    number="08000 156 166"/>
  <UIView
    template="Contact.xml"
    top="previous.bottom + 8.0"
    text="From outside the UK"
    number="(0044) 161 866 4114"/>
</UIView>
```

**Contact.xml**
```xml
<UIView
  width="100%"
  height="auto">
  <param name="text" type="String"/>
  <param name="number" type="String"/>
  <UILabel
    template="Label.xml"
    font="{regularFont}"
    text="{text}"/>
  <UILabel
    template="Label.xml"
    top="64 - height"
    font="{lightFont} 22.0"
    text="{number}"/>
  <UIImageView
    width="64"
    height="64"
    right="100%"
    image="Phone"/>
</UIView>
```

**ClaimsViewController.swift**
```swift
import UIKit
import Layout

class ClaimsViewController: LayoutViewController {
  @objc var localPhoneLabel: UILabel?
  @objc var localPhoneImage: UIImageView?
  @objc var foreignPhoneLabel: UILabel?
  @objc var foreignPhoneLabel: UIImageView?

  // rest of implementation removed for brevity
}
```

What I'd like to be able to do is define 2 outlet parameters on `Contact.xml`—1 for the first label and 1 for the image view—and then set these in `Claims.xml` where `Contact.xml` is embedded. The first embed would be passed the two `localPhone...` outlets, and the second would be passed the two `foreignPhone...` outlets. For instance...

**Claims.xml**
```xml
<UIView>
  <UIView
    template="Contact.xml"
    top="previous.bottom + 8.0"
    text="In the UK"
    number="08000 156 166"
    label-outlet="localPhoneLabel"          <-- New
    image-outlet="localPhoneImage"/>        <-- New
  <UIView
    template="Contact.xml"
    top="previous.bottom + 8.0"
    text="From outside the UK"
    number="(0044) 161 866 4114"
    label-outlet="foreignPhoneLabel"
    image-outlet="foreignPhoneImage"/>
</UIView>
```

**Contact.xml**
```xml
<UIView
  width="100%"
  height="auto">
  <param name="text" type="String"/>
  <param name="number" type="String"/>
  <param name="label-outlet" type="UILabel"/>         <-- New
  <param name="image-outlet" type="UIImageView"/>     <-- New
  <UILabel
    template="Label.xml"
    outlet="{label-outlet}"      <-- New
    font="{regularFont}"
    text="{text}"/>
  <UILabel
    template="Label.xml"
    top="64 - height"
    font="{lightFont} 22.0"
    text="{number}"/>
  <UIImageView
    outlet="{image-outlet}"      <-- New
    width="64"
    height="64"
    right="100%"
    image="Phone"/>
</UIView>
```

Hopefully this makes sense, but feel free to ask for more information if you require it.

**Comment by micpringle** [comment]
Hey @nicklockwood—apologies for the delay in responding to this.

I've just tried the changes and I'm still hitting `Expression for 'outlet' must be  a constant or literal value in UILabel in UIView`?

[AlternateLayout.zip](https://github.com/schibsted/layout/files/1457607/AlternateLayout.zip)

I've attached an updated sample which _I believe_ is set up correctly...

**Comment by nicklockwood** [comment]
@micpringle thanks, I'll take a look.

**Comment by nicklockwood** [comment]
@micpringle OK, it wasn't at all obvious - even to me - but you need to set the type of the outlet params to "String" instead of "UILabel" and "UIImageView".

The reason for this is that the values you are setting for them are keypaths to views that exist in your ViewController - the params don't contain the views themselves. 

I'll see if I can figure out a way to make this more obvious, or at least produce a more helpful error message.

**Comment by micpringle** [comment]
Perfect, thx @nicklockwood!

TBF that makes perfect sense when you think about it, it just wasn't overly obvious. I think a note in the README and a more specific error message would suffice.

Thanks again for looking at this, and my other issues.

**Comment by nicklockwood** [comment]
Fixed in 0.6.0
